confirmation/models.py

```diff
@@ -142,21 +144,25 @@ def create_confirmation_link(
         expiry_date=expiry_date,
         type=confirmation_type,
     )
-    return confirmation_url(key, realm, confirmation_type, url_args)
+    return confirmation_url(key, realm, confirmation_type, url_args, next)
 
 
 def confirmation_url(
     confirmation_key: str,
     realm: Optional[Realm],
     confirmation_type: int,
     url_args: Mapping[str, str] = {},
+    next: str = "",
 ) -> str:
     url_args = dict(url_args)
     url_args["confirmation_key"] = confirmation_key
-    return urljoin(
+    url = urljoin(
         settings.ROOT_DOMAIN_URI if realm is None else realm.uri,
         reverse(_properties[confirmation_type].url_name, kwargs=url_args),
     )
+    if next:
+        url += "?next=" + urllib.parse.quote(next)
```

Comment:
We never want to use `+` to adjust URLs; use the `urrlib` methods like `urljoin`.

This logic may happen to work because we don't expect the URL we've already constructed to contain GET parameters, but we shouldn't rely on that.